packages/datadog-plugin-aws-sdk/src/util.js

```diff
@@ -84,8 +84,67 @@ const extractPrimaryKeys = (keyNames, keyValuePairs) => {
   }
 }
 
+/**
+ * Extracts queue metadata from an SQS queue URL for span tagging.
+ * Handles modern and legacy AWS endpoint formats, with or without schemes.
+ * Automatically detects AWS partitions (standard, China, GovCloud) from region.
+ *
+ * @param {string} queueURL - SQS queue URL in any supported format
+ * @returns {Object|null} Object with queueName and arn, or null if URL format is invalid
+ *
+ * @example
+ * // Modern AWS SQS URLs
+ * extractQueueMetadata('https://sqs.us-east-1.amazonaws.com/123456789012/my-queue')
+ * // Returns { queueName: 'my-queue', arn: 'arn:aws:sqs:us-east-1:123456789012:my-queue' }
+ *
+ * extractQueueMetadata('sqs.eu-west-1.amazonaws.com/123456789012/my-queue') // no scheme
+ * // Returns { queueName: 'my-queue', arn: 'arn:aws:sqs:eu-west-1:123456789012:my-queue' }
+ *
+ * // Legacy AWS SQS URLs
+ * extractQueueMetadata('https://us-west-2.queue.amazonaws.com/123456789012/legacy-queue')
+ * // Returns { queueName: 'legacy-queue', arn: 'arn:aws:sqs:us-west-2:123456789012:legacy-queue' }
+ *
+ * extractQueueMetadata('https://queue.amazonaws.com/123456789012/global-legacy-queue')
+ * // Returns { queueName: 'global-legacy-queue', arn: 'arn:aws:sqs:us-east-1:123456789012:global-legacy-queue' }
+ */
+const extractQueueMetadata = queueURL => {
+  if (!queueURL) {
+    return null
+  }
+
+  const parts = queueURL.split('/').filter(Boolean)
+
+  // Check if URL has scheme
+  const hasScheme = parts[0] && parts[0].startsWith('http')
```

Comment:
Nit: It's just style and now it's consistent a boolean.

```suggestion
  const hasScheme = Boolean(parts[0]?.startsWith('http'))
```